test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs

```diff
@@ -976,6 +977,385 @@ public static CommandRunnerResult RunRestore(SimpleTestPathContext pathContext,
             return r;
         }
 
+        [Fact]
+        public async Task Restore_PackageNamespaceFullPrefixes_Succeed()
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                var net461 = NuGetFramework.Parse("net461");
+
+                var projectA = SimpleTestProjectContext.CreateLegacyPackageReference(
+                    "a",
+                    pathContext.SolutionRoot,
+                    net461);
+                var projectAPackages = Path.Combine(pathContext.SolutionRoot, "packages");
+
+                var externalRepositoryPath = pathContext.PackageSource;
+                var contosoRepositoryPath = pathContext.UserPackagesFolder;
+
+                var configPath = Path.Combine(pathContext.WorkingDirectory, "nuget.config");
+                SettingsTestUtils.CreateConfigurationFile(configPath, $@"<?xml version=""1.0"" encoding=""utf-8""?>
+<configuration>
+    <packageSources>
+    <!--To inherit the global NuGet package sources remove the <clear/> line below -->
+    <clear />
+    <add key=""ExternalRepository"" value=""{externalRepositoryPath}"" />
```

Comment:
Is this is going to pollute your global packages folder?

SimpleTestPathContext creates a nuget.config somewhere in the root directory. 
If you are overwriting that folder, then you could be polluting your own global packages folder.

Reply:
Ok. Replaced with another customer folder.